### PR TITLE
feat: Use github token by default to authenticate download

### PR DESCRIPTION
**Pull request by simse**
I've added a new input `token` which defaults to the one provided by github actions. If no GITHUB_TOKEN env var is set, it will attempt to use the token input. If neither is set, nothing happens.

`GITHUB_TOKEN` takes precedence to provide backwards compatibility.

### CODE REVIEW

**Comment by simse** [comment]
@rossmacarthur voila

File: src/index.ts [hunk 1]
```diff
@@ -3,6 +3,9 @@ import * as setup from "@extractions/setup-crate";
 
 async function main() {
   try {
+    process.env.GITHUB_TOKEN =
+      process.env.GITHUB_TOKEN || core.getInput("token");
+
     const versionSpec = core.getInput("just-version");
     const tool = await setup.checkOrInstallTool({
```
Comment (rossmacarthur) [inline src/index.ts]: Yesterday I published a new version of `@extractions/setup-crate` that accepts this as an argument. Example

```ts
    const tool = await setup.checkOrInstallTool(
      { owner: "casey", name: "just", versionSpec },
      { auth: githubToken }
    );
```

File: action.yaml [hunk 2]
```diff
@@ -7,6 +7,10 @@ branding:
 inputs:
   just-version:
     description: 'A valid semver specifier of the just version to install'
+  token:
```
Comment (rossmacarthur) [inline action.yaml]: Nit: call this `github-token`

**Comment by rossmacarthur** [comment]
@simse any update on this?

**Comment by simse** [comment]
I will address the comments no later than tomorrow evening, don't worry :)

**Comment by simse** [comment]
@rossmacarthur ready for review

**Pull request merged**